### PR TITLE
Updating Wall-E to new SRA Board Component

### PR DESCRIPTION
**Pull request by SuperChamp234**
This PR updates Wall-E to support the new SRA board component, and thus inherently running on the ESP IDF v5.1

The changes include:

1. Removal of switch_control_parallel example as per new SRA Board v2.4
2. Rename of code folders.
3. Adding of mdns idf dependency as it is no longer included in the ESP-IDF by default.
4. Updated gitignore for build folders and files. 
5. Updated submodule to latest sra-board-component.

### CODE REVIEW

File: .gitmodules
Comment (RISHI27-dot) [inline .gitmodules:1]: LGTM

File: 7_self_balancing/main/tuning_websocket_server.c
Comment (RISHI27-dot) [inline 7_self_balancing/main/tuning_websocket_server.c:1]: LGTM

File: 1_led_blink/main/main.c [hunk 1]
```diff
@@ -18,12 +18,12 @@ void app_main()
...
             //0xFF = 1111 1111(all leds are on)
             // setting values of all 8 leds to 1
-            vTaskDelay(1000 / portTICK_RATE_MS);
+            vTaskDelay(1000 / portTICK_PERIOD_MS);
             //delay of 1s
             ESP_ERROR_CHECK(set_bar_graph(0x00));
             //0x00 = 0000 0000(all leds are off)
             // setting values of all 8 leds to 0
-            vTaskDelay(1000 / portTICK_RATE_MS);
+            vTaskDelay(1000 / portTICK_PERIOD_MS);
```
Comment (RISHI27-dot) [inline 1_led_blink/main/main.c:26]: Why is this change required?

File: 6_line_following/main/idf_component.yml
Comment (RISHI27-dot) [inline 6_line_following/main/idf_component.yml:1]: Please specify in the commit message why this file is required.

File: 7_self_balancing/main/idf_component.yml
Comment (RISHI27-dot) [inline 7_self_balancing/main/idf_component.yml:1]: Please specify in the commit message why this file is required.

Comment (SuperChamp234) [reply 7_self_balancing/main/idf_component.yml:1]: https://www.freertos.org/FreeRTOS_Support_Forum_Archive/May_2009/freertos_portTICK_PERIOD_MS_not_portTICK_RATE_MS_3281986.html

FreeRTOS changed the name a few years ago.

File: 6_line_following/main/idf_component.yml
Comment (SuperChamp234) [reply 6_line_following/main/idf_component.yml:1]: https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/protocols/mdns.html

This is the file created when we want to add the mdns dependency. We cannot make a component out of this since it’s a subfolder in a repository. I believe in future we should also switch both sra-board-component and lvgl to the idf registry.

File: 7_self_balancing/main/idf_component.yml
Comment (SuperChamp234) [reply 7_self_balancing/main/idf_component.yml:1]: https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/protocols/mdns.html

This is the file created when we want to add the mdns dependency. We cannot make a component out of this since it’s a subfolder in a repository. I believe in future we should also switch both sra-board-component and lvgl to the idf registry.

File: .gitignore
Comment (RISHI27-dot) [inline .gitignore:1]: I assume this is to ignore files created due to addition of dependency in the idf_component.yml file?

Comment (SuperChamp234) [reply .gitignore:1]: Yes, the temporary build files are not worth committing.

**Review by RISHI27-dot — APPROVED** [review]
